modules/concepts/controllers/admin-controllers/_index.md

```diff
@@ -9,75 +9,179 @@ Using modern pages, you will have access to the PrestaShop debug toolbar, the se
+- **Method parameter injection** for services used in specific actions (`AnotherService`)
+- **Autowire attribute** for injecting services by their ID (`#[Autowire(service: 'my_module.custom_service')]`)
+- **Helper methods** provided by `PrestaShopAdminController` for accessing common PrestaShop services
+
+You have access to Twig as rendering engine and everything from the Symfony framework ecosystem.
+Note that you must return a `Response` object, but this can be a `JsonResponse` if you plan to make a single page application (or "SPA").
+
+{{% notice note %}}
+This controller works exactly the same as the Core Back Office ones.
+{{% /notice %}}
+
+{{% notice info %}}
+**Accessing PrestaShop services:**
+
+The `PrestaShopAdminController` provides helper methods to access common PrestaShop services:
+- `$this->getConfiguration()` - Configuration service
+- `$this->getTranslator()` or `$this->trans()` - Translation service
+- `$this->getRouter()` - Router service
+- `$this->getEmployeeContext()` - Employee context
+- `$this->getShopContext()` - Shop context
+- `$this->getLanguageContext()` - Language context
+- `$this->getCurrencyContext()` - Currency context
+- `$this->getCountryContext()` - Country context
+- `$this->dispatchCommand()` - Execute CQRS commands
+- `$this->dispatchQuery()` - Execute CQRS queries
+- `$this->dispatchHookWithParameters()` - Dispatch hooks
+- `$this->presentGrid()` - Present grid data
+
+See the complete list in the [PrestaShopAdminController source code](https://github.com/PrestaShop/PrestaShop/blob/9.0.0/src/PrestaShopBundle/Controller/Admin/PrestaShopAdminController.php).
+{{% /notice %}}
+
+{{% notice warning %}}
+**Doctrine repositories must be injected:**
+
+Doctrine ORM is not directly accessible via `$this->getDoctrine()` or similar methods. All Doctrine repositories and entity managers must be injected as dependencies in your controller's constructor or action methods.
+
+Example:
+```php
+use Doctrine\ORM\EntityManagerInterface;
+use App\Repository\ProductRepository;
+
+public function __construct(
+    private readonly EntityManagerInterface $entityManager,
+    private readonly ProductRepository $productRepository,
+) {
 }
 ```
+{{% /notice %}} 
+
+## Service Configuration
+
+In PrestaShop 9.0, controllers must be defined as services. You have two main approaches to configure your controller service:
+
+### Option 1: Explicit service configuration with tags
 
-If you want Symfony Dependency Injection to inject services into your controller, you need to use specific YAML service declaration:
 ```yaml
+# modules/your-module/config/services.yml
 services:
-  # The name of the service must match the full namespace class
   MyModule\Controller\DemoController:
-    class: MyModule\Controller\DemoController
-    arguments:
-      - '@doctrine.cache.provider'
+    autowire: true
+    autoconfigure: true
+    tags:
+      - { name: controller.service_arguments }
```

Comment:
I think these ones are doublons, you can either use the tag alon Or the autoconfigure alone but both are doing almost the same thing

So I would favor relying on `autoconfigure: true` and removing the tag part to keep the definition as simple as possible

Reply:
However, regarding the autowiring, maybe we should add an example when you want to inject a service manually But we're entering more in Symfony many possibilities for DI, so instead of redocumenting Symfony we should probably add a notice about the pre-requisite knowledge from the framework itself

The same links that were added in the 9.0 changes pages:
- https://symfony.com/doc/6.4/controller/service.html
- https://symfony.com/doc/6.4/service_container/service_subscribers_locators.html
- https://symfony.com/doc/6.4/service_container.html

and maybe https://symfony.com/doc/6.4/controller.html for a more generic view on what a Symfony controller can do